firedrake/slate/slate.py

```diff
@@ -555,6 +664,23 @@ def arguments(self):
         """Returns a tuple of arguments associated with the tensor."""
         return self._split_arguments
 
+    @cached_property
+    def form(self):
+        tensor, = self.operands
+        assert tensor.terminal
+        if tensor.rank > 1:
+            # turns a Block on a Tensor into an indexed ufl form
+            return ExtractSubBlock().split(tensor.form, self._indices)
+        elif tensor.rank <= 1:
```

Comment:
Can I not have a `tensor.rank == 1` and still have a form not an assembled vector inside?